### PR TITLE
e2e tests: Refactor multihoming tests

### PR DESCRIPTION
**Pull request by oshoval**
**- What this PR does and why is it needed**

Extract code segments into functions.
This will allow also extending the tests and reusing code.
For example can be used by the new tests we are adding at
https://github.com/ovn-org/ovn-kubernetes/pull/4153

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: test/e2e/multihoming.go [hunk 1]
```diff
@@ -14,10 +14,12 @@ import (
 	. "github.com/onsi/gomega"
 
 	"github.com/docker/docker/client"
+	corev1 "k8s.io/api/core/v1"
```
Comment (maiqueb) [inline test/e2e/multihoming.go:17]: this dependency is already present in the line below ...

I agree the original name was unfortunate, and I'd rather have this one, but, IMHO, we should stick to using `v1` now, and later on refactor by renaming the import.

File: test/e2e/multihoming.go [hunk 2]
```diff
@@ -1458,3 +1423,53 @@ func kickstartPod(cs clientset.Interface, configuration podConfiguration) *v1.Po
...
+	By("provisioning the multi-network policy")
+	_, err := mnpClient.MultiNetworkPolicies(namespace).Create(
+		context.Background(),
+		policy,
+		metav1.CreateOptions{},
+	)
+	return err
+}
+
+func getPodNamespace(defaultNamespace string, requiresExtraNamespace bool, extraNamespace string) string {
```
Comment (maiqueb) [inline test/e2e/multihoming.go:1470]: nit: I'd just do this is the caller. This function's signature is IMHO very weird. It is called get pod namespace, does not receive a pod - but the namespaces - plus a boolean.

File: test/e2e/multihoming.go [hunk 3]
```diff
@@ -968,34 +956,17 @@ var _ = Describe("Multi Homing", func() {
...
-					}
-
-					By("sitting on our hands for a couple secs we give the controller time to sync all NADs before provisioning policies and pods")
-					// TODO: this is temporary. We hope to eventually sync pods & multi-net policies on NAD C/U/D ops
-					time.Sleep(3 * time.Second)
+					Expect(createNads(f, nadClient, extraNamespace, netConfig)).NotTo(HaveOccurred())
 
+					kickstartPodInNamespace(cs, &allowedClientPodConfig, f.Namespace.Name, extraNamespace.Name)
+					kickstartPodInNamespace(cs, &blockedClientPodConfig, f.Namespace.Name, extraNamespace.Name)
+					serverPodConfig.namespace = f.Namespace.Name
```
Comment (maiqueb) [inline test/e2e/multihoming.go:963]: hmm I'm surely missing something ... Why can't you use `kickstartPodInNamespace` for the server pod ?

**Review by maiqueb — CHANGES_REQUESTED** [review]
At a glance it looks really good; you're just adding a redundant import, and I have a question plus a nit. 

Thank you.

File: test/e2e/multihoming.go (hunk 3 above)
Comment (oshoval) [reply test/e2e/multihoming.go:963]: nope you are correct, i just was too strict that serverPodConfig doesnt have extra one never
but it is indeed nicer

changing

**Comment by oshoval** [comment]
addressed comments

**Review by maiqueb — APPROVED** [review]
lgtm

Thank you.

**Comment by oshoval** [comment]
There are failures, but not due to this PR
both are on "disabling egress nodes impeding GRCP health check"

**Review by trozet — APPROVED** [review]

**Comment by trozet** [comment]
Flake is https://github.com/ovn-org/ovn-kubernetes/issues/4144

**Pull request merged**